source/rules/ban-observables.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/eslint-plugin-rxjs
+ */
+
+import { stripIndent } from "common-tags";
+import { Rule } from "eslint";
+import * as es from "estree";
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      category: "RxJS",
+      description: "Disallows the use of banned observables.",
+      recommended: false
+    },
+    fixable: null,
+    messages: {
+      forbidden: "RxJS observable is banned: {{name}}{{explanation}}."
+    },
+    schema: [
+      {
+        type: "object",
+        description: stripIndent`
+          An object containing keys that are names of observable factory functions
+          and values that are either booleans or strings containing the explanation for the ban.`
+      }
+    ]
+  },
+  create: context => {
+    let bans: { explanation: string; regExp: RegExp }[] = [];
+
+    const [config] = context.options;
+    if (!config) {
+      return {};
+    }
+
+    Object.entries(config).forEach(([key, value]) => {
+      if (value !== false) {
+        bans.push({
+          explanation: typeof value === "string" ? value : "",
+          regExp: new RegExp(`^${key}$`, "i")
```

Comment:
Added that it could be case insensitive.

Reply:
Was there a particular reason for this? Insensitivity would make sense if it were testing the import locations, but it's testing the imported identifiers. Both `Observable` and `observable` are imports, so this change would make it impossible to differentiate them.

Reply:
Just me not thinking... 😅